depdive/src/update.rs

```diff
@@ -96,6 +97,22 @@ pub enum VersionConflict {
     },
 }
 
+#[derive(Debug, Clone)]
+pub struct VersionUnsafeDiffStats {
+    pub files_changed_containing_unsafe_code: u64,
+    pub files_requiring_review: Vec<String>, // Files potentially containing unsafe change
```

Comment:
This name kind of oversteps. The user of this program presumably has some review policy based on various of the metrics produced, so it seems weird to pick such a policy. Perhaps the previous field should be Vec<String> and you can make an accessor for the count by that just calls len()?

In other words:
```rust
pub struct VersionUnsafeDiffStats {
    pub files_changed_containing_unsafe_code: Vec<String>,
    pub files_unsafe_change_stats: Vec<FileUnsafeChangeStats>,
}
```

Reply:
The way `files_changed_containing_unsafe_code` differs from `files_requiring_review` is that the former is a superset of the latter, additionally containing the count for the files where all unsafe code had been removed in the given diff. But you're right, returning data this way is confusing. We can just keep a `status` enum variant within `FileUnsafeChangeStats` and the user can do all filtering required

Reply:
I was more objecting to part of the name: `_needing_review`. Whether something needs review is very complicated and based on local policy. Your metrics should be objective in what they present as opposed to trying to enforce a policy. Not that you were enforcing a policy, but your naming implied one.